plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -554,9 +555,38 @@ unsafeEvaluateCek runtime = either throw id . extractEvaluationResult . evaluate
 readKnownCek
     :: ( GShow uni, GEq uni, Closed uni, uni `Everywhere` ExMemoryUsage
        , KnownType (Plain Term uni fun) a
-       , Hashable fun, Ix fun, ExMemoryUsage fun
+       , Hashable fun, Ix fun
        )
     => BuiltinsRuntime fun (CekValue uni fun)
     -> Plain Term uni fun
     -> Either (CekEvaluationException uni fun) a
 readKnownCek runtime = evaluateCek runtime >=> readKnown
+
+-- Don't trust this one, use the function from the untyped Cek.
+annotateMemory :: (Everywhere uni ExMemoryUsage, Closed uni) =>
+    Term TyName Name uni fun () -> Term TyName Name uni fun ExMemory
+annotateMemory = cata $ \case
+    VarF () name -> Var (constructorMemoryUsage + memoryUsage name) name
+    TyAbsF () tyname kind term -> TyAbs (constructorMemoryUsage + memoryUsage tyname + toAnnotation kind' + toAnnotation term) tyname kind' term
+        where
+            kind' = withMemory kind
```

Comment:
Doesn't this have the same bug as before for kinds and (below) types? `withMemory` just sets the same annotation on all sub-nodes. Maybe that's fine... but it's not obvious. Worth a comment.

Reply:
Nope, same bug. See above comment on not trusting this function. a) fix it b) mark brokenness more explicitly?

Reply:
Comments are cheap. If it's remotely subtle -> comment.

Reply:
I would argue comments are expensive, because they require upkeep.

Reply:
Sure, but compared to the cost of re-discovering the lost knowledge, it's pretty trivial. How is the next person to look at this function supposed to know which bit is the tricky bit, or that this innocuous line is actually broken?